src/NServiceBus.RabbitMQ.Tests/MessageConverterTests.cs

```diff
@@ -147,36 +146,71 @@ public void TestCanHandleStringHeader()
                     MessageId = "Blah",
                     Headers = new Dictionary<string, object>
                     {
-                        {"Foo", "ni"}
+                        {"short", 42},
+                        {"int", 42},
+                        {"long", 42},
+                        {"decimal", 42m},
+                        {"sbyte", (sbyte)42},
+                        {"double", 42d},
+                        {"single", 42f},
+                        {"bool", true }
                     }
                 }
             };
 
             var headers = converter.RetrieveHeaders(message);
 
             Assert.NotNull(headers);
-            Assert.AreEqual("ni", headers["Foo"]);
+            Assert.AreEqual("42", headers["short"]);
+            Assert.AreEqual("42", headers["int"]);
+            Assert.AreEqual("42", headers["long"]);
+            Assert.AreEqual("42", headers["decimal"]);
+            Assert.AreEqual("42", headers["sbyte"]);
+            Assert.AreEqual("42", headers["double"]);
+            Assert.AreEqual("42", headers["single"]);
+            Assert.AreEqual("True", headers["bool"]);
         }
```

Comment:
> I didn't really see a lot of value in having a bunch of different tests.

While all the cases are passing there's no advantage. When one of the cases fails, the benefit is seen, since the test runner will tell you exactly which case failed. However, given that all of these pass through exactly the same code path, that will never happen, so I'm happy to leave this method as-is. There's probably a small performance benefit too, but perhaps not significant.

> TestCanHandleHeadersWithAllAmqpFieldValues

👍 I like that. With that rename, it makes sense to keep the `decimal` case inside the test.